cookiecutter/generate.py

```diff
@@ -72,18 +72,19 @@ def apply_overwrites_to_context(context, overwrite_context):
             context[variable] = overwrite
 
 
-def generate_context(context_file='cookiecutter.json', default_context=None,
-                     extra_context=None):
+def generate_context(repo_dir, context_file='cookiecutter.json',
```

Comment:
It's intentional: if the parameter's added at the end, then every callsite has to be updated not just with the value of `repo_dir`, but with values for all the other parameters in between when they're not supplied. This simplifies things, since I can ignore whether other callsites are relying on default values.

I'm not too concerned about breaking changes since there's a single callsite in the source code. This mostly affects the test code.